api/environments/dynamodb/dynamodb_wrapper.py

```diff
@@ -161,7 +169,45 @@ def get_item(self, api_key: str) -> dict:
             raise ObjectDoesNotExist() from e
 
 
-class DynamoEnvironmentAPIKeyWrapper(DynamoWrapper):
+class DynamoEnvironmentV2Wrapper(BaseDynamoEnvironmentWrapper):
+    ENVIRONMENT_ID_ATTRIBUTE = "environment_id"
+    DOCUMENT_KEY_ATTRIBUTE = "document_key"
+
+    def get_environment_by_api_key(self, environment_api_key: str) -> dict:
+        filter_expression = Key("environment_api_key").eq(environment_api_key)
+        query_kwargs = {
+            "IndexName": "environment_api_key-index",
+            "Limit": 1,
+            "KeyConditionExpression": filter_expression,
+        }
+        try:
+            return self._table.query(**query_kwargs)["Items"][0]
+        except IndexError:
+            raise ObjectDoesNotExist()
```

Comment:
I've added the sort key to that index so it can be just:

```
.get_item(
    IndexName=ENVIRONMENT_TABLE_V2_SECONDARY_INDEX_NAME,
    Key={
        "environment_api_key": environment_api_key,
        "document_key": "META,
    },
)
```

Reply:
Ah I didn't realise you can do `get_item` with an index.

Reply:
Will update.

Reply:
Done.